src/asm/validate_header.c

```diff
@@ -0,0 +1,116 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   validate_header.c                                  :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: asarandi <asarandi@student.42.us.org>      +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2018/02/11 20:41:20 by asarandi          #+#    #+#             */
+/*   Updated: 2018/03/02 00:41:39 by asarandi         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "asm.h"
+
+void	validate_champion_name(t_asm *a, int *i)
+{
+	int		len;
+	char	*quote1;
+	char	*str;
+
+	str = ft_strdup(a->split[*i]);
+	while ((count_matching_chars(str, '"') < 2) && (*i < a->lines - 1))
+		append_nl_string(&str, a->split[++*i]);
+	if (count_matching_chars(str, '"') > 2)
+		return (header_error(".name has too many quote marks (\")", a));
+	if (count_matching_chars(str, '"') == 2)
+	{
+		quote1 = ft_strchr(str, '"') + 1;
+		len = ft_strchr(quote1, '"') - quote1;
+		if (len < 1)
+			warning(".name is blank");
+		if (len > PROG_NAME_LENGTH)
```

Comment:
See my other comment: Parameterize `PROG_NAME_LENGTH` and this can easily be consolidated.  I do understand the issue about verbosity with errors, but usually someone would more or less figure out that their .name or .comment length is too long.

The compiler errors are generally useless pieces of information, especially when it comes to C++, why worry about that here?